packages/plugin-file/sources/FileResolver.ts

```diff
@@ -42,12 +46,30 @@ export class FileResolver implements Resolver {
   }
 
   async getCandidates(descriptor: Descriptor, dependencies: unknown, opts: ResolveOptions) {
-    let path = descriptor.range;
-
-    if (path.startsWith(PROTOCOL))
-      path = path.slice(PROTOCOL.length);
+    if (!opts.fetchOptions)
+      throw new Error(`Assertion failed: This resolver cannot be used unless a fetcher is configured`);
 
-    return [structUtils.makeLocator(descriptor, `${PROTOCOL}${npath.toPortablePath(path)}`)];
+    const {path, parentLocator} = fileUtils.parseSpec(descriptor.range);
+
+    const folderContents = await fileUtils.loadFolderContents(
+      structUtils.makeLocator(descriptor,
+        structUtils.makeRange({
+          protocol: PROTOCOL,
+          source: path,
+          selector: path,
+          params: {
+            // The Descriptor should already be bound
+            locator: structUtils.stringifyLocator(parentLocator!),
```

Comment:
In the event it isn't, this would be a assertion failure. Just for safety, could you check this and throw if it isn't defined